test/gitea_test.go

```diff
@@ -637,6 +637,76 @@ func TestGiteaProvenance(t *testing.T) {
 	tgitea.WaitForStatus(t, topts, "heads/"+targetRef, "", false)
 }
 
+// TestGiteaGlobalRepoProvenance tests the provenance feature by
+// enabling it for the global repo while disabling it for the local repo.
+// setting pipelinerun_provenance: "default_branch" in global repo.
+func TestGiteaGlobalRepoProvenance(t *testing.T) {
+	topts := &tgitea.TestOpts{
+		SkipEventsCheck:       true,
+		TargetEvent:           triggertype.PullRequest.String(),
+		NoPullRequestCreation: true,
+	}
+
+	ctx := context.Background()
+	topts.ParamsRun, topts.Opts, topts.GiteaCNX, _ = tgitea.Setup(ctx)
+	assert.NilError(t, topts.ParamsRun.Clients.NewClients(ctx, &topts.ParamsRun.Info))
+	ctx, err := cctx.GetControllerCtxInfo(ctx, topts.ParamsRun)
+	assert.NilError(t, err)
+	assert.NilError(t, pacrepo.CreateNS(ctx, topts.TargetNS, topts.ParamsRun))
+
+	globalNs := info.GetNS(ctx)
+	err = tgitea.CreateCRD(ctx, topts,
+		v1alpha1.RepositorySpec{
+			Settings: &v1alpha1.Settings{
+				PipelineRunProvenance: "default_branch",
+			},
+		},
+		true)
+	assert.NilError(t, err)
+
+	defer (func() {
+		if os.Getenv("TEST_NOCLEANUP") != "true" {
+			topts.ParamsRun.Clients.Log.Infof("Cleaning up global repo %s in %s", info.DefaultGlobalRepoName, globalNs)
+			err = topts.ParamsRun.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(globalNs).Delete(
+				context.Background(), info.DefaultGlobalRepoName, metav1.DeleteOptions{})
+			assert.NilError(t, err)
+		}
+	})()
+
+	_, f := tgitea.TestPR(t, topts)
+	defer f()
+	targetRef := topts.TargetRefName
```

Comment:
What will be the value of targetRef here?

Reply:
Its the source branch value

Reply:
OK. I was just trying to understand if it is different from the default branch and it sounds like it is (please correct me if I'm wrong). Thanks.

Reply:
actually current test doesn't check the actualProvenance feature so updating tests

Reply:
I thought it does, that's what I was trying to figure, but now thinking again about it, no, it does not.